### PR TITLE
[ruler] Remove unused fields

### PR DESCRIPTION
**Pull request by gouthamve**
I was helping someone debug notifications being sent every min and we found that
these two fields are not used anywhere at all.

I am not sure of the deprecation and breaking change policy. Should I mark them
deprecated somehow and then remove them or just remove them right away since they
are not being used.

#### Checklist

- [x] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by pracucci — COMMENTED** [review]
Thanks! Could you add a `[CHANGE]` CHANGELOG entry please?

**Comment by gouthamve** [comment]
Done!

**Review by osg-grafana — APPROVED** [review]

**Review by jhesketh — APPROVED** [review]
Change lgtm. Not sure about the deprecation notice process either though.

**Comment by pracucci** [comment]
> LGTM if we're OK just removing these fields. Since they're unused and no longer do anything, I can't imagine anyone has them set?

Yeah, I wouldn't go through the deprecation policy given they're unused.

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,7 @@
 ### Grafana Mimir
 
 * [CHANGE] Compactor: delete source and output blocks from local disk on compaction failed, to reduce likelihood that subsequent compactions fail because of no space left on disk. #2261
+* [CHANGE] Ruler: Remove unused fields `-ruler.search-pending-for` and `-ruler.flush-period`. #2288
```
Comment (pracucci) [inline CHANGELOG.md:8]: [nit] Per docs squad, we don't call them "fields" anymore but we're specific whether they're "CLI flags" or "YAML config options".

```suggestion
* [CHANGE] Ruler: Remove unused CLI flags `-ruler.search-pending-for` and `-ruler.flush-period` (and their respective YAML config options). #2288
```

_Going to auto-apply this edit._

**Pull request merged**